hnn_core/params.py

```diff
@@ -682,18 +678,29 @@ def convert_to_hdf5(params_fname, out_fname,
     None
     """
     from .network import Network
-
+    # Validate inputs
     _validate_type(params_fname, (str, Path), 'params_fname')
     _validate_type(out_fname, (str, Path), 'out_fname')
+    # Convert to Path
+    params_fname = Path(params_fname)
+    out_fname = Path(out_fname)
+
+    params_suffix = params_fname.suffix.lower().split('.')[-1]
+    if params_suffix.lower() not in ['param', 'json']:
```

Comment:
This check is already done in the read Paramus function, and it throws a similar error. Maybe just drop it here?